compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Version.hs

```diff
@@ -67,6 +67,9 @@ featureContractKeys = Feature "Contract keys" version1_3
 featurePartyFromText :: Feature
 featurePartyFromText = Feature "partyFromText function" version1_2
 
+featureComplexContractKeys :: Feature
+featureComplexContractKeys = Feature "Complex contract keys" versionDev
```

Comment:
How will we not screw this up when rolling versionDev? It seems like a likely mistake to make in future.

Reply:
There's a test for the relaxed behavior which only runs for DAML-LF 1.4 and above. Once we roll 1.dev into 1.4 and make it the default version, this test will break because 1.4 does not support complex contract keys. I'll leave comment here and in the test explaining what to do.

Reply:
I wondered about naming versionThatWillBe14 or similar, and using that for versionDev that needs rolling, and only use versionDev for things that will remain versionDev even after a roll.

Reply:
We could also just mark it as 1.4. I assume the default way of cutting a new version of DAML-LF is to roll everything from dev into that version. Being optimistic seems hence justified.